imports/react-ui/settings/knowledgebase/components/topic/topic.js

```diff
@@ -0,0 +1,192 @@
+    return `
+      <script>
+        window.erxesSettings = {
+          messenger: {
+            topic_id: "${topicId}"
+          },
+        };
+        ${KbTopic.installCodeIncludeScript('knowledgeBase')}
+      </script>
+    `;
+  }
+
+  render() {
+    const item = this.props.item || {};
+    const { brands } = this.props;
+
+    return (
+      <form className="margined" onSubmit={this.handleSubmit}>
+        <FormGroup controlId="knowledgebase-title">
+          <ControlLabel>Title</ControlLabel>
+          <FormControl type="text" defaultValue={item.title} required />
+        </FormGroup>
+
+        <FormGroup controlId="knowledgebase-description">
+          <ControlLabel>Description</ControlLabel>
+          <FormControl type="text" defaultValue={item.description} />
+        </FormGroup>
+
+        <FormGroup>
+          <SelectBrand
+            brands={brands}
+            defaultValue={item.brandId}
+            onChange={this.handleBrandChange}
+          />
+        </FormGroup>
+
+        <FormGroup>
+          <ControlLabel>Categories</ControlLabel>
+
+          <Select
+            placeholder="Choose categories"
+            onChange={items => {
+              this.setState({ selectedCategories: items });
+            }}
+            optionRenderer={option => (
+              <div className="simple-option">
+                <span>{option.label}</span>
+              </div>
+            )}
+            value={this.state.selectedCategories}
+            options={this.getCategories()}
+            multi
+          />
+        </FormGroup>
+
+        {this.renderInstallCode()}
+
+        <Modal.Footer>
+          <ButtonToolbar className="pull-right">
+            <Button type="submit" bsStyle="primary">Save</Button>
+          </ButtonToolbar>
+        </Modal.Footer>
+      </form>
+    );
+  }
+
+  getCategories() {
+    const results = [];
+
+    const { categories } = this.props;
+
+    results.push({
+      label: 'Categories',
+      options: categories.map(category => ({
+        label: category.title,
+        value: category._id,
+      })),
+    });
+    return results;
+  }
+
+  renderInstallCode() {
+    if (this.props.item && this.props.item._id) {
+      return (
+        <FormGroup controlId="install-code">
+          <ControlLabel>Install code</ControlLabel>
+          <div className="markdown-wrapper">
+            <ReactMarkdown source={this.state.code} />
+            {this.state.code
+              ? <CopyToClipboard
+                  text={this.state.code}
+                  onCopy={() => this.setState({ copied: true })}
+                >
+                  <Button bsSize="small" bsStyle="primary">
+                    {this.state.copied ? 'Copied' : 'Copy to clipboard'}
+                  </Button>
+                </CopyToClipboard>
+              : null}
+          </div>
+        </FormGroup>
+      );
+    } else {
+      return null;
+    }
+  }
+
+  handleSubmit(e) {
+    super.handleSubmit(e);
```

Comment:
Do we need to close modal after the saving operation is successfully executed?

Reply:
fixed